### ISSUE TITLE
unused-argument is ignored if a function can raise an exception and does not return any value

### ISSUE THREAD

**Issue by gcolombeau** [PyCQA/pylint#3246]
The checker for unused arguments fails to report on functions that do not return anything but contain a raise statement.

### Steps to reproduce
```
"""
unused-argument check.
"""

def my_function(arg_1, *args, arg_2=None):
	"""My function does not use arg."""
	if arg_1 == 42:
		raise ValueError()

````

### Current behavior
No unused arguments are reported.

### Expected behavior
Unused arguments should be reported.

### pylint --version output
pylint 2.4.4
astroid 2.3.3
Python 3.6.6 (default, Sep 25 2019, 09:49:31) 
[GCC 5.4.0 20160609]
